api/internal/managers/app/config/config.go

```diff
@@ -182,58 +189,66 @@ func (m *appConfigManager) GetKotsadmConfigValues() (kotsv1beta1.ConfigValues, e
 			kotsadmConfigValues.Spec.Values[item.Name] = getConfigValueFromItem(item, storedValues)
 
 			for _, childItem := range item.Items {
-				kotsadmConfigValues.Spec.Values[childItem.Name] = getConfigValueFromChildItem(item.Type, childItem, storedValues)
+				kotsadmConfigValues.Spec.Values[childItem.Name] = getConfigValueFromChildItem(item, childItem, storedValues)
 			}
 		}
 	}
 
 	return kotsadmConfigValues, nil
 }
 
-func getConfigValueFromItem(item kotsv1beta1.ConfigItem, configValues types.AppConfigValues) kotsv1beta1.ConfigValue {
-	configValue := kotsv1beta1.ConfigValue{
-		Default: item.Default.String(),
-	}
-	if item.Type == "password" {
-		configValue.ValuePlaintext = item.Value.String()
+// setConfigValueByType sets either Value or ValuePlaintext based on the item type
+func setConfigValueByType(cv *kotsv1beta1.ConfigValue, itemType string, value string) {
+	if itemType == "password" {
+		cv.ValuePlaintext = value
 	} else {
-		configValue.Value = item.Value.String()
+		cv.Value = value
 	}
+}
 
-	// override values from the config values store
-	if v, ok := configValues[item.Name]; ok {
-		if item.Type == "password" {
-			configValue.ValuePlaintext = v.Value
-		} else {
-			configValue.Value = v.Value
-		}
+// applyStoredValueOverride applies stored values from configValues if they exist
+func applyStoredValueOverride(cv *kotsv1beta1.ConfigValue, itemType string, itemName string, configValues types.AppConfigValues) {
```

Comment:
my only comment is that it took my a while to grok what [all of these functions](https://github.com/replicatedhq/embedded-cluster/blob/25c7dc1d229ae0ccaa1d0bf6c068927b847aa245/api/internal/managers/app/config/config.go#L200-L250) do and how the are tied together. is there a way to reduce the number of these functions into maybe 1 or 2 and give them names that are indicative of what’s happening at a higher level? there was only getConfigValueFromChildItem previously i think which was easy to follow.

Reply:
Will add it as follow up.

Reply:
Done via https://github.com/replicatedhq/embedded-cluster/pull/2500